python/cudf/cudf/core/algorithms.py

```diff
@@ -80,6 +83,7 @@ def factorize(
     >>> uniques
     Index([<NA>, 1.0, 2.0], dtype='float64')
     """
+    from cudf.core.index import Index
```

Comment:
Why prefer the local import in this file? We typically don't do that unless it's necessary to avoid circular imports or something like that.

Reply:
Yeah if we want to avoid `import cudf` here (an similar places) we'll need to do this to avoid circular imports.

I'll undo these runtime imports for now and see where things can be restructured so we don't have to resort to these types of imports.

Reply:
Yes function-level imports are slower because you do them every time the functions run. Subsequent runs are generally fast since the modules are already in sys.modules and can be returned immediately, but it's still not completely free.

Reply:
I think restructuring the code in ways that avoids needing circular imports is the better solution long-term. That's what I've asked @brandon-b-miller to do a few times in the past as well.